### PR TITLE
send device category

### PR DESCRIPTION
**Pull request by christiankremserbitmovin**
Set Device Category and save bitrate after stream replay.

### CODE REVIEW

File: src/ts/Html5Metadata.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export class Html5Metadata implements Conviva.MetadataInterface {
 
   // Relying on HTTP user agent string parsing on the Conviva Platform.
   getDeviceCategory(): Conviva.Client.DeviceCategory {
-    return null;
+    return Conviva.Client.DeviceCategory.WEB;
```
Comment (zachmanc) [inline src/ts/Html5Metadata.ts:62]: What’s the expected behavior on smart TVs?

File: src/ts/ConvivaAnalytics.ts [hunk 2]
```diff
@@ -409,6 +409,7 @@ export class ConvivaAnalytics {
     this.client.attachPlayer(this.sessionKey, this.playerStateManager);
 
     if (this.lastSeenBitrate) {
+      this.debugLog('### [ ConvivaAnalytics ] internalInitializeSession, set setBitrateKbps: ', this.lastSeenBitrate);
```
Comment (stonko1994) [inline src/ts/ConvivaAnalytics.ts:412]: We usually don't use `###` in this project.

File: src/ts/ConvivaAnalytics.ts [hunk 3]
```diff
@@ -494,7 +495,8 @@ export class ConvivaAnalytics {
     this.client.releasePlayerStateManager(this.playerStateManager);
 
     this.sessionKey = Conviva.Client.NO_SESSION_KEY;
-    this.lastSeenBitrate = null;
+    // this.lastSeenBitrate = null;
```
Comment (stonko1994) [inline src/ts/ConvivaAnalytics.ts:498]: Please remove commented out code.

Comment (stonko1994) [inline src/ts/ConvivaAnalytics.ts:498]: Why did you remove the reset of the  `lastSeenBitrate`?

File: src/ts/ConvivaAnalytics.ts [hunk 4]
```diff
@@ -615,11 +617,14 @@ export class ConvivaAnalytics {
       // value and use it for tracking when initializing the session.
       // TODO: remove this workaround when the player event order is fixed
       this.lastSeenBitrate = bitrateKbps;
+      // this.debugLog('### [ ConvivaAnalytics ] onVideoQualityChanged, !isSessionActive, set last bitrate: ', this.lastSeenBitrate);
```
Comment (stonko1994) [inline src/ts/ConvivaAnalytics.ts:620]: Please remove commented out code

File: src/ts/ConvivaAnalytics.ts [hunk 5]
```diff
@@ -615,11 +617,14 @@ export class ConvivaAnalytics {
...
       return;
     }
-
-    this.lastSeenBitrate = null;
+    this.lastSeenBitrate = bitrateKbps;
+    // this.lastSeenBitrate = null;
```
Comment (stonko1994) [inline src/ts/ConvivaAnalytics.ts:624]: Please remove commented out code

File: src/ts/ConvivaAnalytics.ts [hunk 6]
```diff
@@ -615,11 +617,14 @@ export class ConvivaAnalytics {
       // value and use it for tracking when initializing the session.
       // TODO: remove this workaround when the player event order is fixed
       this.lastSeenBitrate = bitrateKbps;
+      // this.debugLog('### [ ConvivaAnalytics ] onVideoQualityChanged, !isSessionActive, set last bitrate: ', this.lastSeenBitrate);
       return;
     }
-
-    this.lastSeenBitrate = null;
+    this.lastSeenBitrate = bitrateKbps;
```
Comment (stonko1994) [inline src/ts/ConvivaAnalytics.ts:623]: This is already set in line 619

File: src/ts/Html5Metadata.ts (hunk 1 above)
Comment (stonko1994) [inline src/ts/Html5Metadata.ts:62]: What if you use it on mobile browsers? (Mobile Safari, Chrome, ...) Same for chromecast?

**Review by stonko1994 — CHANGES_REQUESTED** [review]
Please add a changelog entry and unit tests

File: src/ts/Html5Metadata.ts (hunk 1 above)
Comment (christiankremserbitmovin) [reply src/ts/Html5Metadata.ts:62]: Hello, 
this is just a quick fix to set something (it is critical because it should not be empty). There must be of course some device-checks.

File: src/ts/ConvivaAnalytics.ts (hunk 3 above)
Comment (christiankremserbitmovin) [reply src/ts/ConvivaAnalytics.ts:498]: After the video is finished you can click on "replay".  If you set lastSeenBitrate to NULL then the bitrate is empty.

Comment (stonko1994) [reply src/ts/ConvivaAnalytics.ts:498]: Hm. What happens in case of a different source 🤔 Is it ensured that when a new source is loaded the `onVideoQualityChanged` is called?

File: src/ts/Html5Metadata.ts (hunk 1 above)
Comment (stonko1994) [reply src/ts/Html5Metadata.ts:62]: Why not adding those now?

**Comment by mckennajones** [comment]
Closing as this will be handled in #45
